spring-boot-actuator/src/main/java/org/springframework/boot/actuate/health/RedisHealthIndicator.java

```diff
@@ -45,12 +51,30 @@ protected void doHealthCheck(Health.Builder builder) throws Exception {
 				.getConnection(this.redisConnectionFactory);
 		try {
 			Properties info = connection.info();
-			builder.up().withDetail("version", info.getProperty("redis_version"));
+			if (connection instanceof RedisClusterConnection) {
+				redisClusterInfo(builder,
+						((RedisClusterConnection) connection).clusterGetNodes(), info);
+			}
+			else {
+				defaultRedisInfo(builder, VERSION, info.getProperty(REDIS_VERSION));
+			}
 		}
 		finally {
 			RedisConnectionUtils.releaseConnection(connection,
 					this.redisConnectionFactory);
 		}
 	}
 
+	private void defaultRedisInfo(Health.Builder builder, String key, String value) {
+		builder.up().withDetail(key, value);
+	}
+
+	private void redisClusterInfo(Health.Builder builder,
+			Iterable<RedisClusterNode> redisNodes, Properties info) {
+		for (RedisNode redisNode : redisNodes) {
+			defaultRedisInfo(builder, redisNode.asString() + "." + VERSION,
+					info.getProperty(redisNode.asString() + "." + REDIS_VERSION));
```

Comment:
@wilkinsona at the point that I get this data I have only the info properties in the format 
`"<host:port>.<info>":"<value>"`
like
`"127.0.0.1:7002.redis_version":"3.0.7"`
and I don't like the idea of split the string, have you any suggestion to do that ?